vod/hls/m3u8_builder.c

```diff
@@ -809,6 +814,74 @@ m3u8_builder_append_index_url(
 	return p;
 }
 
+static size_t
+m3u8_builder_closed_captions_get_size(
+	media_set_t* media_set,
+	request_context_t* request_context)
+{
+	media_closed_captions_t* closed_captions;
+	size_t result = 0;
+	size_t base;
+
+	base =
+		sizeof(M3U8_EXT_MEDIA_BASE) - 1 +
+		sizeof(M3U8_EXT_MEDIA_TYPE_CLOSED_CAPTIONS) - 1 +
+		sizeof(M3U8_EXT_MEDIA_GROUP_ID_CLOSED_CAPTIONS) - 1 + VOD_INT32_LEN +
+		sizeof(M3U8_EXT_MEDIA_LANG) - 1 +
+		LANG_ISO639_3_LEN +
+		sizeof(M3U8_EXT_MEDIA_INSTREAM_ID) - 1 +
+		sizeof(M3U8_EXT_MEDIA_DEFAULT) - 1 +
+		sizeof("\n") - 1;
+
+	for (closed_captions = media_set->closed_captions; closed_captions < media_set->closed_captions_end; closed_captions++)
+	{
+		result += base + closed_captions->id.len + closed_captions->label.len + sizeof("\n") - 1;
```

Comment:
the `sizeof("\n") - 1` should move to `result` (there's one per cc in `base` + one in global)